cloud-foundry/polices/pas/credhub-key/credhub_key.rego

```diff
@@ -0,0 +1,22 @@
+package credhub
+
+deny_if_not_exactly_one_primary[msg] {
+        keys := [key |
+                input["product-properties"][".properties.credhub_key_encryption_passwords"].value[i].primary # only do the next statement if this is true
+                key := input["product-properties"][".properties.credhub_key_encryption_passwords"].value[i].primary
+        ]
+
+        keys != [true]
+        msg = sprintf("Must have exactly one primary encryption key for credhub, found %d", [count(keys)])
+}
+
+deny_not_enough_chars[msg] {
+        key := [key |
+                input["product-properties"][".properties.credhub_key_encryption_passwords"].value[i].primary # only do the next statement if this is true
+                key := input["product-properties"][".properties.credhub_key_encryption_passwords"].value[i].key.secret
+        ]
+
+        count(key[0]) < 20
+
+        msg = sprintf("Primary key must be at least 20 characters, found %v", [count(key[0])])
```

Comment:
You don't need the comprehension, unless you're trying to eliminate multiple error messages for multiple keys with fewer than 20 characters.
```
deny_not_enough_chars[msg] {
    some i
    input["product-properties"][".properties.credhub_key_encryption_passwords"].value[i].primary
    key := input["product-properties"][".properties.credhub_key_encryption_passwords"].value[i].key.secret
    count(key) < 20
    msg := sprintf("Primary key must be at least 20 characters, found %v", [count(key)])
}
```

Reply:
This was a hot tip 🔥  thank you